From 423b4508c32cbf96446ce4185cdffc655925a43c Mon Sep 17 00:00:00 2001
From: Karthik <quic_karm@quicinc.com>
Date: Tue, 8 Oct 2024 11:37:07 +0530
Subject: [PATCH] QSDK: KW: hostpad: Fix NULL pointer deference

Currently there is no validation check whether memory is allocated
before it is deferenced.

Fix by adding NULL check before accessing the pointer.

Patch-depdendency: none

Patch-work: none

Signed-off-by: Karthik <quic_karm@quicinc.com>
---
 src/eap_server/eap_server_peap.c | 7 ++++++-
 wpa_supplicant/events.c          | 6 ++++++
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/src/eap_server/eap_server_peap.c b/src/eap_server/eap_server_peap.c
index 998d0e8..db5f7a4 100644
--- a/src/eap_server/eap_server_peap.c
+++ a/src/eap_server/eap_server_peap.c
@@ -516,6 +516,11 @@
 	encr_req = eap_server_tls_encrypt(sm, &data->ssl, &msgbuf);
 	os_free(hdr);
 
+	if (encr_req == NULL) {
+		wpa_printf(MSG_INFO, "SSL: Failed to allocate encr_req");
+		return NULL;
+	}
+
 	if (!data->ssl.tls_v13 ||
 	    !tls_connection_resumed(sm->cfg->ssl_ctx, data->ssl.conn)) {
 		wpabuf_free(data->ssl.tls_out);
diff --git a/wpa_supplicant/events.c b/wpa_supplicant/events.c
index 0db7ae2..d4b4ec9 100644
--- a/wpa_supplicant/events.c
+++ b/wpa_supplicant/events.c
@@ -5477,13 +5477,19 @@
 		break;
 	}
 
-	asprintf(&cmd, "CHAN_SWITCH %d %d sec_channel_offset=%d center_freq1=%d center_freq2=%d, bandwidth=%d auto-ht\n",
+	ret = asprintf(&cmd, "CHAN_SWITCH %d %d sec_channel_offset=%d center_freq1=%d center_freq2=%d, bandwidth=%d auto-ht\n",
 		data->ch_switch.count - 1,
 		data->ch_switch.freq,
 		data->ch_switch.ch_offset,
 		data->ch_switch.cf1,
 		data->ch_switch.cf2,
 		width);
+
+	if (ret < 0) {
+		wpa_printf(MSG_ERROR, "\nFailed to allocate memory for cmd");
+		return;
+	}
+
 	ret = wpa_ctrl_request(wpa_s->hostapd, cmd, os_strlen(cmd), buf, &len, NULL);
 	free(cmd);
 
